R/github.R

```diff
@@ -482,3 +583,378 @@ github_api_iterate_pages <- function(token,gh_response,save_folder_path,prefix=N
+    parsed_response[["title"]] <- api_response[["title"]]
+    parsed_response[["body"]] <- api_response[["body"]]
+
+    # Parsing labels
+    parsed_response[["labels"]] <- api_response[["labels"]]
+    if(length(parsed_response[["labels"]]) > 0) {
+      parsed_response[["labels"]] <- stringi::stri_c(sapply(parsed_response[["labels"]], "[[", "name"), collapse = ",")
+    } else {
+      parsed_response[["labels"]] <- NA_character_
+    }
+
+    parsed_response <- as.data.table(parsed_response)
+    return(parsed_response)
+  }
+
+  # Assuming 'items' contains the issues
+  all_issues <- lapply(api_responses[["items"]], parse_response)
+  return(rbindlist(all_issues, fill = TRUE))
+}
+
+
+#' Download Github comment Data by Date
+#'
+#' Appends a 'since' query to the issue/comments api request and returns the first page of the result.
+#'
+#' #' Acceptable formats for `since` are:
+#'
+#' * "YYYY-MM-DD"
+#' * "YYYY-MM-DDTHH:MM"
+#' * "YYYY-MM-DDTHH:MM:SS"
+#' * "YYYY-MM-DDTHH:MM:SSZ"
+#' * "YYYY-MM-DDTHH:MM:SS+00:00"
+#' * NULL
+#'
+#'#' For example: `since="2020-07-04"` (a comment ocurring at the exact specified time will also be downloaded).
+#'
+#' For further details on the `since` Query see [the associated Github API documentation](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#:~:text=asc%2C%20desc-,since,-string).
+#'
+#' @param owner GitHub's repository owner (e.g. sailuh)
+#' @param repo GitHub's repository name (e.g. kaiaulu)
+#' @param token Your GitHub API token
+#' @param since The lower bound. Comments created and/or updated after this date will be retrieved.
+#' @param verbose boolean value. When set to true, it prints operational messages including
+#' greatest dates and the file name that contains the greatest date.
+#' @export
+#' @seealso  \code{link{github_api_project_issue_or_pr_comment_refresh}} to refresh comment data
+#' @seealso  \code{link{github_api_project_issue_refresh}} to refresh issue data
+github_api_project_issue_or_pr_comments_by_date <- function(owner,
+                                                            repo,
+                                                            token,
+                                                            since,
+                                                            verbose = FALSE) {
+  if (is.null(since)) {
+    stop("The lower bound parameter is empty or improperly formatted")
+  }
+  if(verbose){
+    message("Downloading comments updated/created after: ", since)
+  }
+  gh::gh("GET /repos/{owner}/{repo}/issues/comments",
+         owner=owner,
+         repo=repo,
+         since=since,  # Pass the `since` parameter in the API request
+         page=1,
+         per_page=100,
+         .token=token)
+  return(gh_response)
+}
+
+#' Download Github Issue Data by Date
+#'
+#' Appends a 'created' field to a search github JQL query and returns the first page of the response.
```

Comment:
Make it clear that the function is using the created date rather than the updated date endpoint parameter.